Now preview dates uses getfriendlydate format / Fixed a bug with files and folders without creation or modification date

BanCrash opened the pull request:
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #5269

**Details of Changes**

- Fixed a bug that caused files or folders without creation or modification date to show "01/01/1601". For this, I added a new if case to check if the year is 1601 return an empty string, as Windows explorer does:
![image](https://user-images.githubusercontent.com/11770760/122912681-e2c3d600-d358-11eb-8e1d-85308cc9f5ea.png)

- Now preview files and folder uses getfriendlydate format, to fix that bug and to keep consistency while showing dates also.

- Also files and folders properties and details uses getfriendlydate format.

- Fixed a bug that causes that "1 hour" was showed as "1 hours" and "1 minute" as "1 minutes".

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

**Screenshots (optional)**
- Before:
- - File properties of #5269 :
![image](https://user-images.githubusercontent.com/11770760/122912227-629d7080-d358-11eb-860b-e2a97ef7d072.png)

- -  Preview pane:

- - - Bug #5269:
![image](https://user-images.githubusercontent.com/11770760/122912369-8d87c480-d358-11eb-80c0-c17449052ebd.png)

- - - Friendly date on preview:
![image](https://user-images.githubusercontent.com/11770760/122912400-97112c80-d358-11eb-8f8a-727dfa829e61.png)

- After:

- - File properties of #5269 :
![image](https://user-images.githubusercontent.com/11770760/122912039-34b82c00-d358-11eb-9804-9fe7f529f15b.png)

- - Preview pane:

- - - Bug #5269:
![image](https://user-images.githubusercontent.com/11770760/122911649-cc694a80-d357-11eb-9cf0-d8393ee8b9c2.png)

- - - Friendly date on preview:
![image](https://user-images.githubusercontent.com/11770760/122911735-e60a9200-d357-11eb-87cb-568836e2eed9.png)

yaichenbaum commented:
> Now preview files and folder uses getfriendlydate format, to fix that bug and to keep consistency while showing dates also.

@BanCrash Can you make sure the date format follows the setting the user choose in the settings dialog?

BanCrash commented:
> > Now preview files and folder uses getfriendlydate format, to fix that bug and to keep consistency while showing dates also.
> 
> @BanCrash Can you make sure the date format follows the setting the user choose in the settings dialog?

Yep, good point. Added on the last commit.

BanCrash commented:
I just realize this also changes the functionality with properties and details, so I think this PR makes Files to use friendlydate format in every place:

Properties with app date setting:
![image](https://user-images.githubusercontent.com/11770760/122966700-e885df80-d389-11eb-8d29-2739a3ce1b13.png)

Properties with system date setting:
![image](https://user-images.githubusercontent.com/11770760/122966622-d3a94c00-d389-11eb-9205-f2a6095083df.png)

Details on properties with app date setting:
![image](https://user-images.githubusercontent.com/11770760/122966732-f20f4780-d389-11eb-9c64-d44bc85999c9.png)

Details on properties with system date setting:
![image](https://user-images.githubusercontent.com/11770760/122966653-dc9a1d80-d389-11eb-870d-e90a5e61e051.png)



I think this is the proper way, so it keeps consistency showing the dates on every place, but I let you know since I didn't mentioned this on the first post because I didn't knew it.

winston-de commented:
> I think this is the proper way, so it keeps consistency showing the dates on every place, but I let you know since I didn't mentioned this on the first post because I didn't knew it.

I don't think we should show the friendly date format in the preview pane (or details page either). The details in the preview pane are meant to show more detailed information than what's shown in the file list. Same goes for the details page, it is meant to show more precise information.

BanCrash commented:
> > I think this is the proper way, so it keeps consistency showing the dates on every place, but I let you know since I didn't mentioned this on the first post because I didn't knew it.
> 
> I don't think we should show the friendly date format in the preview pane (or details page either). The details in the preview pane are meant to show more detailed information than what's shown in the file list. Same goes for the details page, it is meant to show more precise information.

What about keeping user format choice (system or app), but if they enable app date format then give them the data without relative timespan in those scenarios?

I mean, if creation date is 22/06/2021 23:00, normally it will say "3 minutes ago", but in details or preview it will show "tuesday, 22 june 2021" if the user has choosen app date format, or 22/06/2021 in any part of Files including details or preview if he has choosen the system date format.

winston-de commented:
How about showing both (when the user has the date format set to "application"). For example 06/21/2021 would show as "Wednesday, June 21, 2021 (1 day ago)". When set to system date format, it would just show system date.

BanCrash commented:
> How about showing both (when the user has the date format set to "application"). For example 06/21/2021 would show as "Wednesday, June 21, 2021 (1 day ago)". When set to system date format, it would just show system date.

Sounds good to me, though maybe too much text in that space. In any case I will implement that change and show you the result so you can provide more feedback.

winston-de commented:
Here's a rough mockup. 
![image](https://user-images.githubusercontent.com/59544401/123002446-ca5ab800-d366-11eb-9240-0138821dcfbc.png)

BanCrash commented:
@winston-de done! Here it's the result in spanish:

![image](https://user-images.githubusercontent.com/11770760/123013281-37506b00-d3c4-11eb-976e-c33c6abce50d.png)

I think it's good.

By the way, in the last commit I fixed a bug that caused "1 hour" to be show as "1 hours" and "1 minute" to be show as "1 minutes". Also added the equal operator in the strange case that the days, hours or minutes are exactly the unit (now comparison is >= instead only >).

winston-de reviewed (approved):
LGTM

yaichenbaum commented:
@BanCrash Great work, thank you!

BanCrash commented:
> @BanCrash Great work, thank you!

Perfect, thanks you!
